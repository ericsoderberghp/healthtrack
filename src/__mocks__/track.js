const track = {
  id: 'test-track',
  categories: [
    {
      id: 1,
      name: 'test-yesno',
      aspect: 'behavior',
      type: 'yes/no',
      times: ['16:00'],
    },
    {
      id: 2,
      name: 'test-scale',
      aspect: 'symptom',
      type: 'scale',
      times: ['12:00'],
    },
    {
      id: 3,
      name: 'test-number',
      aspect: 'behavior',
      type: 'number',
      times: ['08:00'],
      units: 'hours',
    },
    {
      id: 4,
      name: 'test-name',
      aspect: 'behavior',
      type: 'name',
      times: ['12:00'],
    },
    {
      id: 5,
      name: 'test-infrequent',
      aspect: 'behavior',
      type: 'scale',
    },
  ],
  createdAt: '2020-06-10T03:50:00.221Z',
  data: [
    {
      id: 1,
      category: 1,
      value: true,
      name: 'test-yesno',
      date: '2020-07-05T23:00:00.451Z',
    },
    {
      id: 2,
      category: 2,
      value: 3,
      name: 'test-scale',
      date: '2020-07-05T23:00:00.792Z',
    },
    {
      id: 3,
      category: 3,
      value: 8,
      name: 'test-number',
      date: '2020-07-05T19:00:00.792Z',
    },
    {
      id: 4,
      category: 4,
      value: 'test-value',
      name: 'test-value',
      date: '2020-07-05T23:00:00.451Z',
    },
    {
      id: 5,
      category: 2,
      value: 2,
      name: 'test-scale',
      date: '2020-07-04T23:00:00.792Z',
    },
  ],
  name: 'Test Track',
  notes: [
    { id: 1, text: 'surfed Grandview', date: '2020-06-20T19:00:00.564Z' },
  ],
  email: 'test@my.com',
};

export default track;
