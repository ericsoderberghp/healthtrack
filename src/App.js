import { Grommet } from 'grommet';
import React, { useEffect, useState } from 'react';
import { Loading } from './components';
import Onboard from './Onboard';
import SignIn from './SignIn';
import Home from './Home';
import Categories from './Categories';
import CategoryAdd from './CategoryAdd';
import CategoryEdit from './CategoryEdit';
import Data from './Data';
import DataAdd from './DataAdd';
import DataEdit from './DataEdit';
import Correlate from './Correlate';
import NoteAdd from './NoteAdd';
import NoteEdit from './NoteEdit';
import Notes from './Notes';
import Router, { Route, Routes } from './Router';
import TrackContext from './TrackContext';
import theme from './theme';
import { useTrack } from './track';

const App = () => {
  const [themeMode, setThemeMode] = useState();
  // align the themeMode with the browser/OS setting
  useEffect(() => {
    if (window.matchMedia) {
      setThemeMode(
        window.matchMedia('(prefers-color-scheme: dark)').matches
          ? 'dark'
          : 'light',
      );
    }
  }, []);

  const [track, setTrack] = useTrack();

  const [nextPath, setNextPath] = useState();
  useEffect(() => {
    if (!nextPath && track !== undefined) {
      // initial load has finished, decide where to go
      if (track) setNextPath(window.location.pathname || '/');
      else if (track === false) setNextPath('/onboard');
    } else if (nextPath !== '/onboard' && track === false) {
      setNextPath('/onboard');
    } else if (nextPath === '/onboard' && track) {
      setNextPath('/');
    }
  }, [nextPath, track]);

  if (!nextPath) return <Loading />;

  return (
    <Router path={nextPath}>
      <Grommet
        theme={theme}
        themeMode={themeMode}
        style={{ minHeight: '100%' }}
      >
        <TrackContext.Provider value={[track, setTrack]}>
          <Routes redirect="/">
            <Route path="/onboard" Component={Onboard} />
            <Route path="/sign-in" Component={SignIn} />
            <Route path="/" Component={Home} />
            <Route path="/categories" Component={Categories} />
            <Route path="/categories/add" Component={CategoryAdd} />
            <Route path="/categories/:id" Component={CategoryEdit} />
            <Route path="/data" Component={Data} />
            <Route path="/data/add" Component={DataAdd} />
            <Route path="/data/:id" Component={DataEdit} />
            <Route path="/correlate" Component={Correlate} />
            <Route path="/notes" Component={Notes} />
            <Route path="/notes/add" Component={NoteAdd} />
            <Route path="/notes/:id" Component={NoteEdit} />
          </Routes>
        </TrackContext.Provider>
      </Grommet>
    </Router>
  );
};

export default App;
