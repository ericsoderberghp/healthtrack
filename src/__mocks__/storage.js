import track from './track';

const store = {};
global.localStorage = {
  getItem: (key) => store[key],
  setItem: (key, value) => (store[key] = value),
};

global.localStorage.setItem('track', JSON.stringify(track));
