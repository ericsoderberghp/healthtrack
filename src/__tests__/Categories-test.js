import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';
import {} from '../__mocks__/storage';

delete window.location;
window.location = new URL('http://localhost/categories');

afterEach(cleanup);

it('renders categories', async () => {
  process.env.NODE_ENV = 'test';
  const { container, findByText } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
  await findByText('test-scale');
  expect(container.firstChild).toMatchSnapshot();
});
