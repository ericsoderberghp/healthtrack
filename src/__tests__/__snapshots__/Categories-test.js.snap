// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders categories 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fZoPPO"
  style="min-height: 100%;"
>
  <div
    class="StyledGrid-sc-1wofa1l-0 jkgEes"
    style="min-height: 100%;"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 kOYHGA"
    >
      <nav
        class="StyledBox-sc-13pk1d4-0 PHNoe"
      >
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          title="calendar"
          type="button"
        >
          <svg
            aria-label="Calendar"
            class="StyledIcon-ofa7kd-0 NqKYz"
            viewBox="0 0 24 24"
          >
            <path
              d="M2,5 L22,5 L22,22 L2,22 L2,5 Z M18,5 L18,1 M6,5 L6,1 M2,10 L22,10"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
        />
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          title="correlate"
          type="button"
        >
          <svg
            aria-label="Test"
            class="StyledIcon-ofa7kd-0 NqKYz"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.9997,0.99999995 L8.9997,8.0003 L1.9997,20.0003 L1.9997,23.0003 L21.9997,23.0003 L21.9997,20.0003 L14.9997,8.0003 L14.9997,0.99999995 M15,18 C15.5522847,18 16,17.5522847 16,17 C16,16.4477153 15.5522847,16 15,16 C14.4477153,16 14,16.4477153 14,17 C14,17.5522847 14.4477153,18 15,18 Z M9,20 C9.55228475,20 10,19.5522847 10,19 C10,18.4477153 9.55228475,18 9,18 C8.44771525,18 8,18.4477153 8,19 C8,19.5522847 8.44771525,20 9,20 Z M18,13 C11,9.99999996 12,17.0000002 6,14 M5.9997,1.0003 L17.9997,1.0003"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
        />
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 jzhQtd"
          kind="default"
          title="categories"
          type="button"
        >
          <svg
            aria-label="Tag"
            class="StyledIcon-ofa7kd-0 NqKYz"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.7058543,22.2941457 C11.3160217,22.6839783 10.6897542,22.6897542 10.2968194,22.2968194 L1.70318058,13.7031806 C1.31482467,13.3148247 1.32097101,12.679029 1.70585426,12.2941457 L13,1 L23,1 L23,11 L11.7058543,22.2941457 Z M6,12 L12,18 M9,9 L15,15 M17,6 C17,5.4475 17.4475,5 18,5 C18.5525,5 19,5.4475 19,6 C19,6.5525 18.5525,7 18,7 C17.4475,7 17,6.5525 17,6"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
        />
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          title="guide"
          type="button"
        >
          <svg
            aria-label="User"
            class="StyledIcon-ofa7kd-0 NqKYz"
            viewBox="0 0 24 24"
          >
            <path
              d="M8,24 L8,19 M16,24 L16,19 M3,24 L3,19 C3,14.0294373 7.02943725,11 12,11 C16.9705627,11 21,14.0294373 21,19 L21,24 M12,11 C14.7614237,11 17,8.76142375 17,6 C17,3.23857625 14.7614237,1 12,1 C9.23857625,1 7,3.23857625 7,6 C7,8.76142375 9.23857625,11 12,11 Z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
      </nav>
      <div
        class="StyledBox-sc-13pk1d4-0 fGNJtn"
      >
        <header
          class="StyledBox-sc-13pk1d4-0 TLWLY"
        >
          <h1
            class="StyledHeading-sc-1rdh4aw-0 bozgir"
          >
            categories
          </h1>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
          />
          <button
            class="StyledButtonKind-sc-1vhfpnt-0 jiUGCi"
            kind="primary"
            title="add category"
            type="button"
          >
            <svg
              aria-label="Add"
              class="StyledIcon-ofa7kd-0 NqKYz"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,22 L12,2 M2,12 L22,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </header>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 fGNJtn"
      >
        <ul
          aria-label="categories"
          class="List__StyledList-sc-130gdqg-0 ckPrwz"
          tabindex="0"
        >
          <li
            class="StyledBox-sc-13pk1d4-0 iSztSp List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-number
            </span>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
            />
            8 am
          </li>
          <li
            class="StyledBox-sc-13pk1d4-0 excDPH List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-name
            </span>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
            />
            12 pm
          </li>
          <li
            class="StyledBox-sc-13pk1d4-0 excDPH List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-scale
            </span>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
            />
            12 pm
          </li>
          <li
            class="StyledBox-sc-13pk1d4-0 excDPH List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-yesno
            </span>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
            />
            4 pm
          </li>
          <li
            class="StyledBox-sc-13pk1d4-0 excDPH List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-infrequent
            </span>
            
          </li>
        </ul>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 iTGgII"
      >
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 gPerFl"
          >
            <svg
              aria-label="User"
              class="StyledIcon-ofa7kd-0 NqKYz"
              viewBox="0 0 24 24"
            >
              <path
                d="M8,24 L8,19 M16,24 L16,19 M3,24 L3,19 C3,14.0294373 7.02943725,11 12,11 C16.9705627,11 21,14.0294373 21,19 L21,24 M12,11 C14.7614237,11 17,8.76142375 17,6 C17,3.23857625 14.7614237,1 12,1 C9.23857625,1 7,3.23857625 7,6 C7,8.76142375 9.23857625,11 12,11 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
            />
            guide
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders categories 2`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fZoPPO"
  style="min-height: 100%;"
>
  <div
    class="StyledGrid-sc-1wofa1l-0 jkgEes"
    style="min-height: 100%;"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 kOYHGA"
    >
      <nav
        class="StyledBox-sc-13pk1d4-0 PHNoe"
      >
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          title="calendar"
          type="button"
        >
          <svg
            aria-label="Calendar"
            class="StyledIcon-ofa7kd-0 NqKYz"
            viewBox="0 0 24 24"
          >
            <path
              d="M2,5 L22,5 L22,22 L2,22 L2,5 Z M18,5 L18,1 M6,5 L6,1 M2,10 L22,10"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
        />
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          title="correlate"
          type="button"
        >
          <svg
            aria-label="Test"
            class="StyledIcon-ofa7kd-0 NqKYz"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.9997,0.99999995 L8.9997,8.0003 L1.9997,20.0003 L1.9997,23.0003 L21.9997,23.0003 L21.9997,20.0003 L14.9997,8.0003 L14.9997,0.99999995 M15,18 C15.5522847,18 16,17.5522847 16,17 C16,16.4477153 15.5522847,16 15,16 C14.4477153,16 14,16.4477153 14,17 C14,17.5522847 14.4477153,18 15,18 Z M9,20 C9.55228475,20 10,19.5522847 10,19 C10,18.4477153 9.55228475,18 9,18 C8.44771525,18 8,18.4477153 8,19 C8,19.5522847 8.44771525,20 9,20 Z M18,13 C11,9.99999996 12,17.0000002 6,14 M5.9997,1.0003 L17.9997,1.0003"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
        />
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 jzhQtd"
          kind="default"
          title="categories"
          type="button"
        >
          <svg
            aria-label="Tag"
            class="StyledIcon-ofa7kd-0 NqKYz"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.7058543,22.2941457 C11.3160217,22.6839783 10.6897542,22.6897542 10.2968194,22.2968194 L1.70318058,13.7031806 C1.31482467,13.3148247 1.32097101,12.679029 1.70585426,12.2941457 L13,1 L23,1 L23,11 L11.7058543,22.2941457 Z M6,12 L12,18 M9,9 L15,15 M17,6 C17,5.4475 17.4475,5 18,5 C18.5525,5 19,5.4475 19,6 C19,6.5525 18.5525,7 18,7 C17.4475,7 17,6.5525 17,6"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
        />
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          title="guide"
          type="button"
        >
          <svg
            aria-label="User"
            class="StyledIcon-ofa7kd-0 NqKYz"
            viewBox="0 0 24 24"
          >
            <path
              d="M8,24 L8,19 M16,24 L16,19 M3,24 L3,19 C3,14.0294373 7.02943725,11 12,11 C16.9705627,11 21,14.0294373 21,19 L21,24 M12,11 C14.7614237,11 17,8.76142375 17,6 C17,3.23857625 14.7614237,1 12,1 C9.23857625,1 7,3.23857625 7,6 C7,8.76142375 9.23857625,11 12,11 Z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
      </nav>
      <div
        class="StyledBox-sc-13pk1d4-0 fGNJtn"
      >
        <header
          class="StyledBox-sc-13pk1d4-0 TLWLY"
        >
          <h1
            class="StyledHeading-sc-1rdh4aw-0 bozgir"
          >
            categories
          </h1>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
          />
          <button
            class="StyledButtonKind-sc-1vhfpnt-0 jiUGCi"
            kind="primary"
            title="add category"
            type="button"
          >
            <svg
              aria-label="Add"
              class="StyledIcon-ofa7kd-0 NqKYz"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,22 L12,2 M2,12 L22,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
        </header>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 fGNJtn"
      >
        <ul
          aria-label="categories"
          class="List__StyledList-sc-130gdqg-0 ckPrwz"
          tabindex="0"
        >
          <li
            class="StyledBox-sc-13pk1d4-0 iSztSp List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-number
            </span>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
            />
            8 am
          </li>
          <li
            class="StyledBox-sc-13pk1d4-0 excDPH List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-name
            </span>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
            />
            12 pm
          </li>
          <li
            class="StyledBox-sc-13pk1d4-0 excDPH List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-scale
            </span>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
            />
            12 pm
          </li>
          <li
            class="StyledBox-sc-13pk1d4-0 excDPH List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-yesno
            </span>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
            />
            4 pm
          </li>
          <li
            class="StyledBox-sc-13pk1d4-0 excDPH List__StyledItem-sc-130gdqg-1 fBniOQ"
            tabindex="-1"
          >
            <span
              class="StyledText-sc-1sadyjn-0 gmtNWw"
            >
              test-infrequent
            </span>
            
          </li>
        </ul>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 iTGgII"
      >
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 gPerFl"
          >
            <svg
              aria-label="User"
              class="StyledIcon-ofa7kd-0 NqKYz"
              viewBox="0 0 24 24"
            >
              <path
                d="M8,24 L8,19 M16,24 L16,19 M3,24 L3,19 C3,14.0294373 7.02943725,11 12,11 C16.9705627,11 21,14.0294373 21,19 L21,24 M12,11 C14.7614237,11 17,8.76142375 17,6 C17,3.23857625 14.7614237,1 12,1 C9.23857625,1 7,3.23857625 7,6 C7,8.76142375 9.23857625,11 12,11 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
            <div
              class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
            />
            guide
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
